azurekeyVaultPollSample/src/main/java/com/microsoft/runAzureKeyVaultPoller.java

```diff
@@ -60,104 +54,106 @@ public static void main( String[] args )
-     * @param kvClient instance of Azure KeyVaultClient
+     * @param tokenCredential Use it to build KeyClient,secretClient
      * @param vaultBaseUrl the url of the vault where the secret and keys are stored
      */
-    public static void runSample(KeyVaultClient kvClient, String vaultBaseUrl){
-
-        try {
-
-            //Example of how to create a key using KV client, in the specified vault
-            KeyBundle keyRSA = kvClient.createKey(new CreateKeyRequest.Builder(vaultBaseUrl,"keyRSA", JsonWebKeyType.RSA).build());
-            System.out.println("The key Id is: " + keyRSA.key().kid());
-
-            //Example: Async example of creating secret in specified vault, null passed for callback
-            ServiceFuture<SecretBundle> secretAsync = kvClient.setSecretAsync(new SetSecretRequest.Builder(vaultBaseUrl,"secretNameInVault","secretValue").build(),null);
-            System.out.println("The secret value is: " + secretAsync.get().value());
-
-            System.out.println("Now listing keys and secrets in Azure Key Vault.");
-
-            for(KeyItem ki: kvClient.listKeys(vaultBaseUrl,10)){ //list keys in vault up to a maximum of 10 records
-                System.out.println("key tag is: " + ki.tags());
-                System.out.println("key attributes: " + ki.attributes());
-                String key_url = ki.kid();
-                System.out.println("key id: " + key_url);
-
-                runSigning(kvClient,key_url);
-
-                KeyBundle keyBundle = kvClient.getKey(key_url);
-                String kid = keyBundle.key().kid(); // get the key identifier
-                System.out.println("Key in Key Vault: " + kid);
+    public static void runSample(TokenCredential tokenCredential, String vaultBaseUrl, Vault vault) throws InterruptedException {
+
+        KeyClient keyClient = new KeyClientBuilder().credential(tokenCredential).vaultUrl(vaultBaseUrl).buildClient();
+        SecretClient secretClient = new SecretClientBuilder().credential(tokenCredential).vaultUrl(vaultBaseUrl).buildClient();
+        SecretAsyncClient secretAsyncClient = vault.secretClient();
+
+        //Example of how to create a key using keyClient, in the specified vault
+        KeyVaultKey savedKey = keyClient.createKey("keyRSA", KeyType.RSA);
+        System.out.println("The key Id is: " + savedKey.getId());
+
+        //Example: Async example of creating secret in specified vault, null passed for callback
+        secretAsyncClient.setSecret("secretNameInVault", "secretValue").subscribe(keyVaultSecret -> {
+            System.out.println("The secret value is: " + keyVaultSecret.getValue());
+        });
+
+        Thread.sleep(10000);
+
+        System.out.println("Now listing keys and secrets in Azure Key Vault.");
+
+        keyClient.listPropertiesOfKeys().stream().forEach(keyProperty -> {
+            System.out.printf("key attributes : enabled is %s, notBefore is %s, expires is %s, created is %s, updated is %s \n",
```

Comment:
nit: should the printf more closely match the property names?  specificially expires -> ExpiresOn, created -> CreatedOn, and updated -> UpdatedOn

Reply:
According to your comment, it has been fixed in the new version .